pkg/machines/components/vmDisksTab.jsx

```diff
@@ -106,13 +107,27 @@ const VmDisksTab = ({ idPrefix, vm, disks, actions, renderCapacity, dispatch, pr
                     columns.push(disk.diskExtras);
 
                     if (provider === 'LibvirtDBus') {
-                        const removeDiskAction = RemoveDiskAction({
-                            dispatch,
-                            vm,
-                            target: disk.target,
-                            idPrefixRow,
-                            onAddErrorNotification,
-                        });
+                        const onDetachDisk = () => {
+                            return () => {
+                                dispatch(detachDisk({ connectionName:vm.connectionName, id:vm.id, name:vm.name, target: disk.target, live: vm.state == 'running' }))
+                                        .catch(ex => {
+                                            onAddErrorNotification({
+                                                text: cockpit.format(_("Disk $0 fail to get detached from VM $1"), disk.target, vm.name),
+                                                detail: ex.message, resourceId: vm.id,
+                                            });
+                                        })
+                                        .then(() => dispatch(getVm({ connectionName: vm.connectionName, id:vm.id })));
+                            };
+                        };
+
+                        const removeDiskAction = <Delete objectType="Disk"
+                                   className='machines-listing-actions'
+                                   objectName={disk.target}
+                                   objectId={"disk" + disk.target}
+                                   disabled={vm.state != 'shut off' && vm.state != 'running'}
+                                   overlayText={_("The VM needs to be running or shut off to detach this device")}
+                                   actionName={_("Detach")}
```

Comment:
This changes the current deisgn, to show Detach string in the button instead of [-] that used to be before.
So it needs design review from @andreasn or @garrett